### PR TITLE
Make sure data rows don't show html formatting in excel exports [PNA]

### PR DESCRIPTION
**Pull request by pr33thi**
[[Interrupt ticket]](https://dimagi-dev.atlassian.net/secure/RapidBoard.jspa?rapidView=5&projectKey=HI&modal=detail&selectedIssue=HI-347&quickFilter=3)

My original fix (https://github.com/dimagi/commcare-hq/pull/23203) only addressed the `total_rows`, which is the row at the very bottom of the excel spreadsheet (and all I saw when I reproduced this- my tables don't have data locally). This covers all the data rows as well. 

@czue

### CODE REVIEW

File: custom/intrahealth/utils.py [hunk 1]
```diff
@@ -363,20 +363,28 @@ def export_table(self):
...
 
+    def _extract_all_rows_from_report(self, report_table):
+        extracted_rows = []
+        for row in report_table['rows']:
+            extracted_rows.append(self._extract_single_row_from_report(row))
+        return extracted_rows
+
     @staticmethod
-    def _extract_value_from_report_table_row_value(report_table):
+    def _extract_single_row_from_report(list_to_extract_from):
```
Comment (mkangia) [inline custom/intrahealth/utils.py]: can this be called `row` instead of `list_to_extract_from`?

Comment (mkangia) [reply custom/intrahealth/utils.py]: how about `_sanitize_row`?

**Review by mkangia — APPROVED** [review]
changes look fine. Naming can use another thought

File: custom/intrahealth/tests/reports/test_dashboard_1.py [hunk 2]
```diff
@@ -25,7 +25,10 @@ def test_extract_value_from_report_table_row_value_input_dict(self):
         report_table = {
             'fix_column': False,
             'comment': 'test_comment',
-            'rows': [],
+            'rows': [[{'html': '0'}, {'html': '4'}, {'html': '8'}, {'html': '12'}],
+                     [{'html': '1'}, {'html': '5'}, {'html': '9'}, {'html': '13'}],
+                     [{'html': '2'}, {'html': '6'}, {'html': '10'}, {'html': '14'}],
+                     [{'html': '3'}, {'html': '7'}, {'html': '11'}, {'html': '15'}]],
```
Comment (mkangia) [inline custom/intrahealth/tests/reports/test_dashboard_1.py]: Do we want to check some example with funny characters like in the original PR `{u'style': u'', u'html': u'pas de donn\xe9es'}`

Comment (pr33thi) [reply custom/intrahealth/tests/reports/test_dashboard_1.py]: those are way better names, thanks https://github.com/dimagi/commcare-hq/pull/23245/commits/6bfa1926fd18985c93518e56247dcff8c943dfdc

File: custom/intrahealth/utils.py (hunk 1 above)
Comment (pr33thi) [reply custom/intrahealth/utils.py]: those are way better names, thanks https://github.com/dimagi/commcare-hq/pull/23245/commits/6bfa1926fd18985c93518e56247dcff8c943dfdc

**Review by mkangia — APPROVED** [review]

**Pull request merged**